Project.toml

```diff
@@ -47,7 +48,8 @@ LinearAlgebra = "37e2e46d-f89d-539d-b4ee-838fcccc9c8e"
 NLPModels = "a4795742-8479-5a88-8948-cc11e1c8c1a6"
 NLPModelsJuMP = "792afdf1-32c1-5681-94e0-d7bf7a5df49e"
 OptimizationProblems = "5049e819-d29b-5fba-b941-0eee7e64c1c6"
+PartitionedVectors = "9d5e22db-ddc8-46c9-8309-91b035e7f733"
```

Comment:
If a package is in the `deps` section, you don't need it in the `extras`/`targets` sections.

Reply:
> Thanks @paraynaud that's a huge change ! That looks good overall. I proposed a solution for the typing.

Thanks for the review @tmigot !